diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d742f82..1c0d2b5e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,11 +3,13 @@ project(rostime)
 
 find_package(catkin REQUIRED COMPONENTS cpp_common)
 find_package(Boost REQUIRED COMPONENTS date_time system thread)
+find_library(RT_LIBRARIES rt)
+
 catkin_package(
   INCLUDE_DIRS include
   LIBRARIES ${PROJECT_NAME}
   CATKIN_DEPENDS cpp_common
-  DEPENDS Boost
+  DEPENDS Boost RT
 )
 
 include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
