diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c367c9e..2b775f37 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,8 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /permissive-")
+
 # Define global caktin_lint suppressions
 #catkin_lint: ignore uninstalled_script missing_install_target
 
@@ -27,7 +29,14 @@ find_package(urdfdom_headers REQUIRED)
 
 find_package(PkgConfig REQUIRED)
 
+set(MSVC_TOOLSET_VERSION "141")
 find_package(ASSIMP QUIET)
+unset(MSVC_TOOLSET_VERSION)
+
+
+add_definitions(-DSTRICT)
+add_definitions(-DQ_WINSTRICT)
+
 if (NOT ASSIMP_FOUND)
   pkg_check_modules(ASSIMP assimp)
 endif()
diff --git a/src/rviz/CMakeLists.txt b/src/rviz/CMakeLists.txt
index bfde3bca..b143e6a8 100644
--- a/src/rviz/CMakeLists.txt
+++ b/src/rviz/CMakeLists.txt
@@ -1,5 +1,7 @@
 find_package(yaml-cpp REQUIRED)
 
+find_library(YAML_CPP_LINK_LIBRARIES yaml-cpp)
+
 include_directories(${YAML_CPP_INCLUDE_DIR})
 
 if(UNIX AND NOT APPLE)
@@ -143,7 +145,7 @@ target_link_libraries(${PROJECT_NAME}
   ${TinyXML2_LIBRARIES}
   ${X11_X11_LIB}
   ${ASSIMP_LIBRARIES}
-  ${YAML_CPP_LIBRARIES}
+  ${YAML_CPP_LINK_LIBRARIES}
 )
 
 add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
diff --git a/src/rviz/default_plugin/interactive_markers/interactive_marker_control.h b/src/rviz/default_plugin/interactive_markers/interactive_marker_control.h
index fd435e5d..0dfb5ee6 100644
--- a/src/rviz/default_plugin/interactive_markers/interactive_marker_control.h
+++ b/src/rviz/default_plugin/interactive_markers/interactive_marker_control.h
@@ -32,6 +32,11 @@
 
 
 #ifndef Q_MOC_RUN
+#ifdef _WIN32
+#undef ERROR
+#undef DELETE
+#endif
+
 #include <boost/shared_ptr.hpp>
 #include <boost/enable_shared_from_this.hpp>
 
diff --git a/src/rviz/default_plugin/marker_display.cpp b/src/rviz/default_plugin/marker_display.cpp
index 0240c13d..23bc7836 100644
--- a/src/rviz/default_plugin/marker_display.cpp
+++ b/src/rviz/default_plugin/marker_display.cpp
@@ -45,6 +45,10 @@
 
 #include "rviz/default_plugin/marker_display.h"
 
+#ifdef DELETE
+#undef DELETE
+#endif
+
 namespace rviz
 {
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////
diff --git a/src/rviz/default_plugin/marker_utils.cpp b/src/rviz/default_plugin/marker_utils.cpp
index 5aa179a0..e07dd456 100644
--- a/src/rviz/default_plugin/marker_utils.cpp
+++ b/src/rviz/default_plugin/marker_utils.cpp
@@ -45,6 +45,10 @@
 #include "rviz/validate_quaternions.h"
 #include "rviz/validate_floats.h"
 
+#ifdef DELETE
+#undef DELETE
+#endif
+
 namespace rviz
 {
 MarkerBase*
diff --git a/src/rviz/ogre_helpers/render_system.cpp b/src/rviz/ogre_helpers/render_system.cpp
index e04bfa74..95bad147 100644
--- a/src/rviz/ogre_helpers/render_system.cpp
+++ b/src/rviz/ogre_helpers/render_system.cpp
@@ -160,10 +160,11 @@ void RenderSystem::setupDummyWindowId()
 
 void RenderSystem::loadOgrePlugins()
 {
-  std::string plugin_prefix = get_ogre_plugin_path() + "/";
-#ifdef Q_OS_MAC
-  plugin_prefix += "lib";
-#endif
+  #ifdef _WIN32
+  std::string plugin_prefix = std::string(std::getenv("CONDA_PREFIX")) + "\\Library\\bin\\";
+  #else
+  std::string plugin_prefix = std::string(std::getenv("CONDA_PREFIX")) + "/lib/OGRE/";
+  #endif
   ogre_root_->loadPlugin(plugin_prefix + "RenderSystem_GL");
   ogre_root_->loadPlugin(plugin_prefix + "Plugin_OctreeSceneManager");
   ogre_root_->loadPlugin(plugin_prefix + "Plugin_ParticleFX");
diff --git a/src/rviz/yaml_config_writer.cpp b/src/rviz/yaml_config_writer.cpp
index 0ef7fe53..1f3175e1 100644
--- a/src/rviz/yaml_config_writer.cpp
+++ b/src/rviz/yaml_config_writer.cpp
@@ -152,7 +152,9 @@ void YamlConfigWriter::writeConfigNode(const Config& config, YAML::Emitter& emit
     break;
   }
   default:
-    emitter << YAML::Null;
+  #ifndef _WIN32
+    emitter << YAML::_Null;
+  #endif
     break;
   }
 }
