diff --git a/CMakeLists.txt b/CMakeLists.txt
index c418afce..1b142d70 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,16 +1,9 @@
 cmake_minimum_required(VERSION 3.0.2)
 project(tf)
 
-include(CheckCXXCompilerFlag)
-unset(COMPILER_SUPPORTS_CXX11 CACHE)
-if(MSVC)
-  # https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version
-  # MSVC has c++14 enabled by default, no need to specify c++11
-else()
-  check_cxx_compiler_flag(-std=c++11 COMPILER_SUPPORTS_CXX11)
-  if(COMPILER_SUPPORTS_CXX11)
-    add_compile_options(-std=c++11)
-  endif()
+ # Melodic default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
 
 find_package(catkin REQUIRED COMPONENTS
diff --git a/include/tf/tf.h b/include/tf/tf.h
index 934aabf6..f51624ff 100644
--- a/include/tf/tf.h
+++ b/include/tf/tf.h
@@ -60,6 +60,12 @@
   #define TF_DECL
 #endif
 
+// Boost winapi.h includes winerror.h. Subsequently NO_ERROR gets defined
+// and which conflicts with tf::NO_ERROR.
+#if defined(_WIN32) && defined(NO_ERROR)
+  #undef NO_ERROR
+#endif
+
 namespace tf
 {
 /** \brief resolve tf names */
diff --git a/src/tf.cpp b/src/tf.cpp
index ac9ad0da..13f027c4 100644
--- a/src/tf.cpp
+++ b/src/tf.cpp
@@ -30,7 +30,6 @@
 /** \author Tully Foote */
 
 #include "tf/tf.h"
-#include <sys/time.h>
 #include "ros/assert.h"
 #include "ros/ros.h"
 #include <angles/angles.h>
