diff --git a/CMakeLists.txt b/CMakeLists.txt
index 41a3b7b..c0f57de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,13 +147,14 @@ ENDIF(G2O_USE_OPENMP)
 
 # OpenGL is used in the draw actions for the different types, as well
 # as for creating the GUI itself
-FIND_PACKAGE(OpenGL)
-SET(G2O_USE_OPENGL ON CACHE BOOL "Build g2o with OpenGL support for visualization")
-IF (OPENGL_FOUND AND G2O_USE_OPENGL)
-  SET (G2O_HAVE_OPENGL 1)
-  INCLUDE_DIRECTORIES(${OPENGL_INCLUDE_DIR})
-  MESSAGE(STATUS "Compiling with OpenGL support")
-ENDIF()
+# FIND_PACKAGE(OpenGL)
+# SET(G2O_USE_OPENGL ON CACHE BOOL "Build g2o with OpenGL support for visualization")
+# IF (OPENGL_FOUND AND G2O_USE_OPENGL)
+#   SET (G2O_HAVE_OPENGL 1)
+#   INCLUDE_DIRECTORIES(${OPENGL_INCLUDE_DIR})
+#   INCLUDE_DIRECTORIES(${CMAKE_INSTALL_PREFIX}/include)
+#   MESSAGE(STATUS "Compiling with OpenGL support")
+# ENDIF()
 
 # For building the GUI
 FIND_PACKAGE(QGLViewer)
